expose local web server using ngrok

lzboy14 opened the pull request:
Notion Ticket Link:

https://www.notion.so/uwblueprintexecs/e2d9163a2f8c436bbb36172066753d40?v=9518aa1af04b4478b3d5e840f4ffcc7f&p=850e4246e7d74065826f1de8203d5457&pm=s

Summary:
- Added ngrok implementation so devs have a public URL to hit
- Tested backend endpoints with cURL
- Will install ngrok on docker compose up --build
- Devs must register for their own ngrok account and add their authtoken as an environment variable
- Run ngrok http http://localhost:5001 to open the ngrok interface

Expected result:
1. Rebuilding docker container and running docker compose up --build
2. Run ngrok http http://localhost:5001 to open the ngrok interface (public URL logged in terminal)
3. Public URL should be able to be accessed with cURL GET and POST requests

Ongoing/Incomplete:
* Still need to do a thorough cURL testing of the endpoints (ik how to do them now - will be done by june 5th, 2024)
* Also need to figure out authorization for hitting endpoints

Questions/Notes for Matt:
- In the Dockerfile in the backend / typescript folder, I changed the EXPOSE line to expose 5001 instead of 5000 (line 27)
- It was previously exposing 5000, but that didn't seem to have an effect, so I just changed it to 5001.
- I set most things related to the backend to port 5001 instead of 5000 (see changes)
- I'm not sure how to run a linter (see checklist below)?

## Checklist
- [y] My PR name is descriptive and in imperative tense
- [y] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [?] I have run the appropriate linter(s)
- [y] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

mat-ng commented:
Great work! I tested it out with ngrok and the port forwarding works! Let me know if you guys need any help with the remaining work you mentioned (API testing, etc.). I left a few other PR comments for you guys to address.

Additionally, I'll answer some of the questions/notes you left for me:
> In the Dockerfile in the backend / typescript folder, I changed the EXPOSE line to expose 5001 instead of 5000 (line 27)
> It was previously exposing 5000, but that didn't seem to have an effect, so I just changed it to 5001.

Thanks for doing that. You're right, the `EXPOSE` instruction just helps document which port the app _intends_ to use (for documentation and other purposes), but doesn't actually publish to the port. As such, you're correct about it not "affecting anything" and it was good to change it to `5001` so it's consistent throughout our repo.

> I set most things related to the backend to port 5001 instead of 5000 (see changes)

Thanks for making this change too. I remember from the first work session that a good number of people had issues with their port 5000, so we can just switch to running our backend on port 5001 for convenience reasons.

> I'm not sure how to run a linter (see checklist below)?

In our first work session, remember we ran a linter that checked our repo for formatting, etc. (see [here](https://github.com/uwblueprint/don-efficace?tab=readme-ov-file#linter)). Just run that linter again to ensure that your changes don't have any additional linting errors. You can ignore the existing linting errors caused by other files in the repo.

lzboy14 commented:
1. Ran the linter (which is why some of the unrelated files got minor changes in indentation, spacing, etc.)
2. Deleted the ngrok container in docker-compose.yml
3. Deleted .DS_Store file

mat-ng reviewed (approved):
Very nice, this is going to be massively helpful for some upcoming tickets!
